crates/cranelift/src/func_environ.rs

```diff
@@ -507,14 +524,23 @@ impl<'module_environment> FuncEnvironment<'module_environment> {
         let fuel = builder
             .ins()
             .load(ir::types::I64, ir::MemFlags::trusted(), addr, offset);
-        builder.def_var(self.fuel_var, fuel);
+
+        if self.tunables.outband_fuel {
+            builder.ins().set_pinned_reg(fuel);
+        } else {
+            builder.def_var(self.fuel_var, fuel);
+        }
     }
 
     /// Stores the fuel consumption value from `self.fuel_var` into
     /// `VMRuntimeLimits`.
     fn fuel_save_from_var(&mut self, builder: &mut FunctionBuilder<'_>) {
         let (addr, offset) = self.fuel_addr_offset(builder);
-        let fuel_consumed = builder.use_var(self.fuel_var);
+        let fuel_consumed = if self.tunables.outband_fuel {
+            builder.ins().get_pinned_reg(ir::types::I64)
```

Comment:
Similar to `fuel_load_into_var` I think this function may never be executed if `outband_fuel` is true?